### PR TITLE
Upgrade to reCAPTCHA v2

### LINKED ISSUE #26: Upgrade Recaptcha to V2 - $150US bountysource
I'm wondering if there are any plans to update this to V2 now that it has been released?

I started looking into the code to see if I could to see if I could upgrade the code, but I'm not too sure I follow the code very well.

### PR DESCRIPTION
**Pull request by edlinklater**
This upgrades to v2 of the Google reCAPTCHA API. Noscript support (via iframe) is maintained.

As per Google's documentation, HTTPS is always used.

Fixes #26

### CODE REVIEW

**Comment by HAZARDU5** [comment]
Looks like the CI tests for this merge request are failing against PHP 5.3 but are fine in PHP 5.4. Is there something PHP 5.4 specific you're using in your code?

**Comment by edlinklater** [comment]
@HAZARDU5: I'd expect it would actually be failing everywhere, as the tests aren't valid any longer due to a major change in the way reCAPTCHA v2 works. Am planning to discuss this with @chillu when he's back in the office.

File: code/RecaptchaField.php [hunk 1]
```diff
@@ -155,79 +90,55 @@ public function Field($properties=array()) {
...
-	public function getJsOptionsString() {
-		$js = "{";
-		$i=1;
-		foreach($this->jsOptions as $k => $v) {
-			$js .= "'{$k}':'{$v}'";
-			if($i < count($this->jsOptions)) $js .= ',';
-			$i++;
-		}
-		$js .= "}";
+		$html .= <<<EOF
```
Comment (HAZARDU5) [inline code/RecaptchaField.php]: This block is encapsulated by `<<<EOF`. Is this intentional? It could be causing PHP 5.3 CI tests to fail.

Comment (edlinklater) [inline code/RecaptchaField.php]: @HAZARDU5: Heredoc syntax isn't new. If you check the [failures](https://travis-ci.org/chillu/silverstripe-recaptcha/jobs/71346588) you'll see they're all due to "Undefined index: challenge". The challenge/response works very differently in reCAPTCHA v2 so the tests will need rewriting.

Comment (HAZARDU5) [inline code/RecaptchaField.php]: Ah cool :-)

We were previously using this for a client's site so we'll look some workarounds to get it up and running with v2 in the meantime.

Comment (edlinklater) [inline code/RecaptchaField.php]: @HAZARDU5: It works fine as-is (for me at least), just the tests aren't checking the right things. As I said I'll discuss this with @chillu once he's back in the office, but in the meantime I'm using it on a project.

File: README.md [hunk 2]
```diff
@@ -60,60 +60,12 @@ Configuration example in `mysite/_config/spamprotection.yml`
 
 Then once you have setup this config you will need to include the spam protector field as per the instructions on the [spamprotection](https://github.com/silverstripe/silverstripe-spamprotection) page.
 
-
-## Custom Theming
```
Comment (chillu) [inline README.md]: Why has this been removed? It's still a feature of the API: https://developers.google.com/recaptcha/old/docs/customization#custom-theming

File: code/RecaptchaField.php [hunk 3]
```diff
@@ -155,79 +90,55 @@ public function Field($properties=array()) {
...
-			if(!empty($this->jsOptions) && isset($this->jsOptions['theme']) && $this->jsOptions['theme'] === 'custom'){
-				$html .= $this->renderWith("CustomRecaptchaField");
-			}
-			$html .= '
-				<script type="text/javascript" src="' . $jsURL . '">
-				</script>
-			';
+
+		// turn options array into data attributes
+		$option_string = '';
```
Comment (chillu) [inline code/RecaptchaField.php]: Coding conventions - should be camel case

File: README.md (hunk 2 above)
Comment (chillu) [inline README.md]: Ah forget it, they've put the old v1 docs on the same nav as the v2 docs ... functionality doesn't exist in v2 any more.

**Comment by chillu** [comment]
Looking good overall, thanks Ed! Can you please fix that one coding convention thing, and document required upgrading steps in the commit message better? For example, explicitly state that `useAjaxAPI` has been removed and won't have an effect any more.

**Comment by PatchRanger** [comment]
@edlinklater Please take a look at backer's post at Bountysource: https://www.bountysource.com/issues/23983531-upgrade-recaptcha-to-v2-150us-bountysource (the last one at the moment) - I guess, you should complete this pull-request (I mean make it merged) in order to receive a bounty.

**Comment by zarocknz** [comment]
Hi Guys, any progress on this since October last year? Would be nice to use recapture v2

**Comment by chillu** [comment]
Replaced by https://github.com/chillu/silverstripe-recaptcha/pull/36

**Pull request closed without merging**